projects/sitemanage/src/main/java/com/percussion/sitemanage/task/impl/PSSiteMapGeneratorTask.java

```diff
@@ -0,0 +1,159 @@
+     * @param endTime   the time when the job completed, before the first post
+     *                  task is invoked. This time is only available to post tasks and
+     *                  will be <code>null</code> for pre edition tasks.
+     * @param jobId     the job id.
+     * @param duration  the length of time that the edition ran in seconds, from
+     *                  the first moment <i>after</i> the pre tasks completed to the
+     *                  moment just before the first post edition tasks started.
+     *                  Supplied as <code>0</code> to pre edition tasks.
+     * @param success   if <code>true</code> then the edition was successful,
+     *                  which means that all items published without error. If
+     *                  <code>false</code> then some or all items failed and the
+     *                  status callback or other service calls must be used to
+     *                  determine what failures existed. Undefined for pre edition
+     *                  tasks.
+     * @param params    registered parameters for the task, may be empty or
+     *                  <code>null</code> for tasks that don't require parameters.
+     * @param status    this is <code>null</code> for pre tasks, but post tasks
+     *                  can use this to obtain status information about the job.
+     * @throws Exception if the task fails for any reason an exception should
+     *                   be thrown that details the reason for the failure. The
+     *                   exception will be caught by the job code and recorded as part
+     *                   of the edition task.
+     */
+    @Override
+    public void perform(IPSEdition edition, IPSSite site, Date startTime, Date endTime, long jobId, long duration, boolean success, Map<String, String> params, IPSEditionTaskStatusCallback status) throws Exception {
+
+        List<IPSEditionContentList> contentLists = publisherService.loadEditionContentLists(edition.getGUID());
+
+        if(!success){
+            log.warn("Skipping sitemap generation for failed job: {}", jobId);
+            return;
+        }
+
+        //Detect if this is a "normal" edition - publish now, auto publish, and incremental should always be skipped.
+        if(contentLists != null && ! contentLists.isEmpty()){
+            for(IPSEditionContentList ecl : contentLists){
+                IPSContentList cl = publisherService.loadContentList(ecl.getContentListId());
+                if(cl.getType() == IPSContentList.Type.INCREMENTAL){
+                    //If this is an incremental edition we want to skip generation
+                    log.warn("Skipping sitemap generation for incremental edition. ");
+                    return;
+                }
+
+                if(cl.getGenerator().equalsIgnoreCase("Java/global/percussion/system/sys_SelectedItemsGenerator")){
+                    log.warn("Skipping sitemap generation for Publish Now edition.");
```

Comment:
can cl.getGenerator() ever be null?

Reply:
I don't think it can here, but I can add a check to guard that.